### PR TITLE
fix(compiler): errors do not show inflight in type name (attempt 2)

### LINKED ISSUE #1501: Confusing compiler error when inflight keyword is missing
### Community Note

> Please vote by adding a 👍 reaction to the issue to help us prioritize.
> If you are interested to work on this issue, please leave a comment.

### I tried this:

```
bring cloud;

new cloud.Function((name: str): str => {
    return "Hello ${name}";
});
```

### I expected this:

An error message telling me that the function I passed is not inflight

### Instead, this happened:

Expected type to be "fn(any): any", but got "fn(str): str" instead

### Component:

Compiler

### Wing Version:

0.5.23

### Node.js Version:

18.12.1

### Platform(s):

MacOS

### Anything else?

We should show the word `inflight` as part of the type (i.e. `inflight fn(any): any`)

### PR DESCRIPTION
**Pull request by Chriscbr**
Fixes #1501

*By submitting this pull request, I confirm that my contribution is made under the terms of the [Monada Contribution License](https://docs.winglang.io/terms-and-policies/contribution-license.html)*.

### CODE REVIEW

**Comment by Chriscbr** [comment]
There's a bit of duplication because we have a `Type` enum in the `ast` module and a `Type` enum in the `type_checker` module. (I changed the `Display` trait on only one of them, hence why the bug wasn't fixed). This doesn't seem ideal, but I couldn't think of a clean way to avoid the duplication. do you have thoughts on this @yoav-steinberg?

File: libs/wingc/src/ast.rs [hunk 1]
```diff
@@ -117,32 +117,36 @@ impl Display for Type {
...
-					.collect::<Vec<String>>()
-					.join(", ");
-				let ret_type_str = if let Some(ret_val) = &sig.return_type {
-					format!("{}", ret_val)
-				} else {
-					"void".to_string()
-				};
-				write!(f, "{phase_str}({params_str}): {ret_type_str}",)
-			}
+			Type::FunctionSignature(sig) => write!(f, "{}", sig),
```
Comment (MarkMcCulloh) [inline libs/wingc/src/ast.rs:120]: I don't feel like this is duplication, our naming just makes it seem like it.
One of these corresponds to the AST node for a function type annotation, the other is our representation of a function type in our type system. 

The `Type` used in this file should probably be renamed to `TypeAnnotation` or something to help with that.

**Review by MarkMcCulloh — APPROVED** [review]
Nice fix, just noticed the phase was missing in the type text a minute ago too 🎉

File: libs/wingc/src/ast.rs (hunk 1 above)
Comment (Chriscbr) [reply libs/wingc/src/ast.rs:120]: Ah great idea, I like `TypeAnnotation` 👍

**Pull request merged**